libraries/AP_RangeFinder/AP_RangeFinder_uLanding.h

```diff
@@ -24,11 +24,17 @@ class AP_RangeFinder_uLanding : public AP_RangeFinder_Backend
     }
 
 private:
+    // detect uLanding Firmware Version
+    bool detect_uLanding_version(void);
+
     // get a reading
     bool get_reading(uint16_t &reading_cm);
 
     AP_HAL::UARTDriver *uart = nullptr;
     uint32_t last_reading_ms = 0;
-    uint8_t linebuf[10];
+    uint8_t linebuf[6];
     uint8_t linebuf_len = 0;
+    bool _uLanding_version_check = false;
+    uint8_t _uLanding_hdr;
+    uint8_t _uLanding_version;
```

Comment:
I think we can lose the uLanding part, it's already the driver name. Also, any reason these variables are prefixed by underscore when the previous one aren't? We don't have a rule on prefixing or not, but keeping the same style in a class is definitely wanted.

Reply:
Oversight on my part. I'm used to generally following the [style guide on private members](http://ardupilot.org/dev/docs/style-guide.html#private-members), and the linebuf variables were leftovers from the initial version of the driver.

Reply:
After another look, I realized the linebuf and linebuf_len variables are only used in the get_reading() function, so I moved them there. That also allowed a little cleanup with the buffer size (which is dependent on the _version number) and the check against the linebuf_len while filling the buffer.